controllers/articles-controllers.js

```diff
@@ -17,7 +18,11 @@ exports.getArticles = async (req, res, next) => {
 exports.getArticleById = async (req, res, next) => {
   try {
     const { article_id: id } = req.params;
-    const article = await fetchArticle(id);
+    const [article, { rows: comments }] = await Promise.all([
+      fetchArticle(id),
+      fetchCommentsByArticleId(id),
```

Comment:
The idea is that you query the database using a JOIN to comments. 

The foreign key should line up in comments to article id, then you can use aggregate functions to count the comments. 
At the moment, you're bypassing a lot of the PQSL functionality and the point of having a relational database.